org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/internal/workspace/ProjectConfigurators.java

```diff
@@ -0,0 +1,132 @@
+
+        SubMonitor progress = SubMonitor.convert(monitor);
+        progress.setWorkRemaining(this.contributions.size());
+        for (ProjectConfiguratorContribution contribution : this.contributions) {
+            try {
+                contribution.getConfigurator().configure(context, progress.newChild(1));
+            } catch (Exception e) {
+                logFailureAndAddErrorMarker(contribution, e, "configure project '" + project.getName() + "'");
+            }
+        }
+    }
+
+    void unconfigureConfigurators(IProject project, IProgressMonitor monitor) {
+        ProjectContext context = newProjectContext(project);
+
+        SubMonitor progress = SubMonitor.convert(monitor);
+        progress.setWorkRemaining(this.contributions.size());
+        for (ProjectConfiguratorContribution contribution : this.contributions) {
+            try {
+                contribution.getConfigurator().unconfigure(context, progress.newChild(1));
+            } catch (Exception e) {
+                logFailureAndAddErrorMarker(contribution, e, "unconfigure project '" + project.getName() + "'");
+            }
+        }
+    }
+
+    static ProjectConfigurators create(GradleBuild gradleBuild, List<ProjectConfiguratorContribution> configurators) {
+        return new ProjectConfigurators(gradleBuild, configurators);
+    }
+
+    private static InitializationContext newInitializationContext(GradleBuild internalGradleBuild) {
+        BuildConfiguration buildConfiguration = internalGradleBuild.getBuildConfig().toApiBuildConfiguration();
+        org.eclipse.buildship.core.GradleBuild gradleBuild = GradleCore.getWorkspace().createBuild(buildConfiguration);
+        return new InitializationContext() {
+
+            @Override
+            public org.eclipse.buildship.core.GradleBuild getGradleBuild() {
+                return gradleBuild;
+            }
+        };
+    }
+
+    private static ProjectContext newProjectContext(IProject project) {
+        return new ProjectContext() {
+
+            @Override
+            public IProject getProject() {
+                return project;
+            }
+        };
+    }
+
+    private void logFailureAndAddErrorMarker(ProjectConfiguratorContribution contribution, Exception exception, String operationName) {
+        String message = String.format("Project configurator '%s' from plugin '%s' failed to %s",
+                contribution.getConfigurator().getClass().getSimpleName(),
+                contribution.getContributorPluginId(),
+                operationName);
+        Status status = new Status(IStatus.WARNING, contribution.getContributorPluginId(), message, exception);
+        CorePlugin.getInstance().getLog().log(status);
+
+        try {
+            IResource targetLocation = markerLocation();
+            GradleErrorMarker.create(targetLocation, this.gradleBuild, message, exception, 0);
+        } catch (CoreException e) {
+            CorePlugin.getInstance().getLog().log(e.getStatus());
```

Comment:
❌ This is still logging using our logger, thus AERI reports will be sent to Buildship. We should log using the logger of the faulty configurator.